pkg/frontend/admin_openshiftcluster_kubernetesobjects_test.go

```diff
@@ -235,11 +235,13 @@ func TestAdminKubernetesObjectsGetAndDelete(t *testing.T) {
 			openshiftClusters := mock_database.NewMockOpenShiftClusters(controller)
 			tt.mocks(tt, openshiftClusters, kactions)
 
-			f, err := NewFrontend(ctx, logrus.NewEntry(logrus.StandardLogger()), _env, &database.Database{
+			f, err := New(ctx, logrus.NewEntry(logrus.StandardLogger()), _env, &database.Database{
 				OpenShiftClusters: openshiftClusters,
-			}, api.APIs, &noop.Noop{}, nil, func(*logrus.Entry, env.Interface) kubeactions.Interface {
-				return kactions
-			}, nil, nil, nil)
+			}, api.APIs, &noop.Noop{}, nil,
+				WithKubeActionsFactory(func(*logrus.Entry, env.Interface) kubeactions.Interface {
```

Comment:
I'm not sure if this will work but can you pass in `pkg/frontend/kubeactions.New` _(which has the same function signature)_?  

Because the `WithkubeActionsFactory` function sets the `kubeActionsFactory:` in the struct, wouldn't passing in the `kubeActions.New` satisfy this? Or 
```bash
WithKubeActionsFactory() 
```
which will set that in the function.

```bash
func WithKubeActionsFactory() Option {
	return func(f *frontend) {
		f.kubeActionsFactory = kubeactions.New
	}
}
```

Reply:
We can't use `kubeactions.New` here because we need the function to return `kactions`, a mock, which was previously created on line 234. Calls to `kactions` will be expected, so we really need that exact, preexisting object to be returned.